src/unittest/vg.cpp

```diff
@@ -143,55 +144,888 @@ TEST_CASE("is_acyclic() should return whether the graph is acyclic", "[vg][cycle
+            }
+            else if (orig_id == 3 && !flipped && n.sequence() == graph.get_node(orig_id)->sequence()) {
+                node_5 = n.id();
+            }
+            else if (orig_id == 3 && flipped && n.sequence() == reverse_complement(graph.get_node(orig_id)->sequence())) {
+                node_10 = n.id();
+            }
+            else if (orig_id == 4 && !flipped && n.sequence() == graph.get_node(orig_id)->sequence()) {
+                node_6 = n.id();
+            }
+            else if (orig_id == 4 && flipped && n.sequence() == reverse_complement(graph.get_node(orig_id)->sequence())) {
+                node_7 = n.id();
+            }
+            else if (orig_id == 5 && !flipped && n.sequence() == graph.get_node(orig_id)->sequence()) {
+                node_8 = n.id();
+            }
+            else if (orig_id == 5 && flipped && n.sequence() == reverse_complement(graph.get_node(orig_id)->sequence())) {
+                node_9 = n.id();
+            }
+        }
+        
+        REQUIRE(node_1 != 0);
+        REQUIRE(node_2 != 0);
+        REQUIRE(node_3 != 0);
+        REQUIRE(node_4 != 0);
+        REQUIRE(node_5 != 0);
+        REQUIRE(node_6 != 0);
+        REQUIRE(node_7 != 0);
+        REQUIRE(node_8 != 0);
+        REQUIRE(node_9 != 0);
+        REQUIRE(node_10 != 0);
+        
+        bool found_edge_1 = false;
+        bool found_edge_2 = false;
+        bool found_edge_3 = false;
+        bool found_edge_4 = false;
+        bool found_edge_5 = false;
+        bool found_edge_6 = false;
+        bool found_edge_7 = false;
+        bool found_edge_8 = false;
+        bool found_edge_9 = false;
+        bool found_edge_10 = false;
+        bool found_edge_11 = false;
+        bool found_edge_12 = false;
+        
+        for (int i = 0; i < g.edge_size(); i++) {
+            const Edge& e = g.edge(i);
+            if ((e.from() == node_1 && e.to() == node_4 && !e.from_start() && !e.to_end()) ||
+                (e.from() == node_4 && e.to() == node_1 && e.from_start() && e.to_end())) {
+                found_edge_1 = true;
+            }
+            else if ((e.from() == node_4 && e.to() == node_5 && !e.from_start() && !e.to_end()) ||
+                     (e.from() == node_5 && e.to() == node_4 && e.from_start() && e.to_end())) {
+                found_edge_2 = true;
+            }
+            else if ((e.from() == node_9 && e.to() == node_6 && !e.from_start() && !e.to_end()) ||
+                     (e.from() == node_6 && e.to() == node_9 && e.from_start() && e.to_end())) {
+                found_edge_3 = true;
+            }
+            else if ((e.from() == node_6 && e.to() == node_5 && !e.from_start() && !e.to_end()) ||
+                     (e.from() == node_5 && e.to() == node_6 && e.from_start() && e.to_end())) {
+                found_edge_4 = true;
+            }
+            else if ((e.from() == node_5 && e.to() == node_3 && !e.from_start() && !e.to_end()) ||
```

Comment:
As much as I love your thorough test code, maybe this could be compressed through the use of a loop?